sdk/eventhub/event-hubs/src/eventHubConsumerClient.ts

```diff
@@ -425,13 +441,15 @@ export class EventHubConsumerClient {
     }
 
     eventProcessor.start();
+    this._eventProcessors.add(eventProcessor);
```

Comment:
Just curious - is it a deliberate choice to store off the EventProcessor and not the returned Subscription object?

If you use the returned object you get the .delete() from the this._eventProcessors for free (it's here and in your close above)

Reply:
The thought did cross my mind after I pushed the changes. I can make this change.

Reply:
Updated!